code-challenges.js

```diff
@@ -31,38 +83,223 @@ const temperature2 = 350
+ * @return {string} The reversed string
+ * 
+ * @example reverseString('hello') // --> 'olleh'
+ * @throws {TypeError} str is not a string
+ */
+const reverseString = (str) => {
+    if (typeof str !== 'string') {
+        throw new TypeError('a string must be provided')
+    }
+
+    return str.split("").reverse().join('')
+}
+
+console.log(reverseString(currentCohort)) // --> '3202 ovarB'
+// console.log(reverseString(5)) // --> TypeError: a string must be provided
+// console.log(reverseString()) // --> TypeError: a string must be provided
+
 // --------------------4) Create the code that will return the last index of the given value from the array using the test variables provided below.
 
 // Pseudo code:
 
+/*
+Create a function that takes an array and a number or string
+Use Array.prototype.lastindexof to return the last index of the the given number or string
+*/
+
 const myNumbers = [13, 34, 42, 5, 5, 10, 27, 42, 10]
 const givenValue1 = 42
 // Expected output: 7
 
 const givenValue2 = 10
 // Expected output: 8
 
+/**
+ * Get the last index of a given item
+ * @param {*[]} arr - array to search
+ * @param {*} item - item to search for
+ * @return {number} index of last occurance, -1 if not found
+ * 
+ * @example getLastIndexOf([1, 2, 3, 4, 5, 4], 4) // --> 5
+ * @throws {TypeError} arr is not an array
+ */
+const getLastIndexOf = (arr, item) => {
+    if (!(arr instanceof Array)) {
+        throw new TypeError('arr is not an array')
+    }
+
+    return arr.lastIndexOf(item)
+}
+
+console.log(getLastIndexOf(myNumbers, givenValue1)) // --> 7
+console.log(getLastIndexOf(myNumbers, givenValue2)) // --> 8
+// console.log(getLastIndexOf([1, 2, 3, 4, 5, 4], 4)) // --> 5
+// console.log(getLastIndexOf(["hey", "hello", "yo", 4, 5, "hello", "hello", 3], "hello")) // --> 6
+// console.log(getLastIndexOf([5, 6, 7, 4, true], "hello")) // --> -1
+// console.log(getLastIndexOf([5, 6, 7, 4, true, false, 6], true)) // --> 4
+
 // --------------------5) Create the code that will sort all the numbers in an array from largest to smallest. Use the test variables provided below.
 
 // Pseudo code:
 
+/*
+Create a function that accepts an array
+return the array using the sort method and reverse to start with largest items
+*/
+
 const sanDiegoSummerTemperatures = [79, 80, 72, 73, 82, 77, 76]
 // Expected output: [82, 80, 79, 77, 76, 73, 72]
 
 const sanDiegoWinterTemperatures = [59, 68, 62, 59, 66, 67, 66]
 // Expected output: [68, 67, 66, 66, 62, 59, 59]
+
+/**
+ * Sort an array of numbers from largest to smallest 
+ * @param {number[]} arr - array to sort
+ * @returns {number[]} sorted array
+ * 
+ * @example sortFromLargestToSmallest([1, 2, 3, 4, 5, 4]) // --> [5, 4, 4, 3, 2, 1]
+ * @throws {TypeError} arr is not an array of numbers
+ */
+const sortFromLargestToSmallest = (arr) => {
+    if (arr.some(isNaN)) {
+        throw new TypeError('arr is not an array of numbers')
+    }
+
+    return arr.sort().reverse()
+}
```

Comment:
Absolutely crushed this section! We can pass a function to the sort method that will change the way items are sorted if you want to look into that.